Tests/RemoteMessagingTests/RemoteMessagingStoreTests.swift

```diff
@@ -270,126 +256,114 @@ class RemoteMessagingStoreTests: XCTestCase {
     }
 
-    func testWhenFeatureFlagIsDisabledThenFetchedMessageReturnsNil() throws {
-        let remoteMessage = try saveProcessedResultFetchRemoteMessage()
+    func testWhenFeatureFlagIsDisabledThenUpdateShownFlagHasNoEffect() async throws {
+        let remoteMessage = try await saveProcessedResultFetchRemoteMessage()
+        try await setFeatureFlagEnabled(false)
 
-        availabilityProvider.isRemoteMessagingAvailable = false
-
-        XCTAssertNil(store.fetchRemoteMessage(withID: remoteMessage.id))
-    }
-
-    func testWhenFeatureFlagIsDisabledThenUpdateShownFlagHasNoEffect() throws {
-        let remoteMessage = try saveProcessedResultFetchRemoteMessage()
-        availabilityProvider.isRemoteMessagingAvailable = false
-
-        store.updateRemoteMessage(withID: remoteMessage.id, asShown: true)
+        await store.updateRemoteMessage(withID: remoteMessage.id, asShown: true)
 
         XCTAssertFalse(store.hasShownRemoteMessage(withID: remoteMessage.id))
     }
 
-    func testWhenFeatureFlagIsDisabledThenHasShownMessageReturnFalse() throws {
-        let remoteMessage = try saveProcessedResultFetchRemoteMessage()
-        store.updateRemoteMessage(withID: remoteMessage.id, asShown: true)
-        availabilityProvider.isRemoteMessagingAvailable = false
+    func testWhenFeatureFlagIsDisabledThenHasShownMessageReturnFalse() async throws {
+        let remoteMessage = try await saveProcessedResultFetchRemoteMessage()
+        await store.updateRemoteMessage(withID: remoteMessage.id, asShown: true)
+        try await setFeatureFlagEnabled(false)
 
         XCTAssertFalse(store.hasShownRemoteMessage(withID: remoteMessage.id))
     }
 
-    func testWhenFeatureFlagIsDisabledThenDismissingRemoteMessageHasNoEffect() throws {
-        let remoteMessage = try saveProcessedResultFetchRemoteMessage()
+    func testWhenFeatureFlagIsDisabledThenDismissingRemoteMessageHasNoEffect() async throws {
+        let remoteMessage = try await saveProcessedResultFetchRemoteMessage()
 
-        availabilityProvider.isRemoteMessagingAvailable = false
-        store.dismissRemoteMessage(withID: remoteMessage.id)
+        try await setFeatureFlagEnabled(false)
+        await store.dismissRemoteMessage(withID: remoteMessage.id)
 
         XCTAssertEqual(store.fetchDismissedRemoteMessageIDs(), [])
     }
 
-    func testWhenFeatureFlagIsDisabledThenHasDismissedRemoteMessageReturnsFalse() throws {
-        let remoteMessage = try saveProcessedResultFetchRemoteMessage()
-
-        store.dismissRemoteMessage(withID: remoteMessage.id)
-        availabilityProvider.isRemoteMessagingAvailable = false
+    // MARK: -
 
-        XCTAssertEqual(store.hasDismissedRemoteMessage(withID: remoteMessage.id), false)
+    func setFeatureFlagEnabled(_ isRemoteMessagingAvailable: Bool) async throws {
+        availabilityProvider.isRemoteMessagingAvailable = isRemoteMessagingAvailable
+        try await Task.sleep(interval: 0.1)
```

Comment:
This helper function updates the feature flag and waits for a bit in order to allow the store to delete the currently scheduled message. That happens asynchronously and we can't easily learn when it finished.